server/jetstream_cluster_2_test.go

```diff
@@ -6391,41 +6388,88 @@ func TestJetStreamClusterRePublishUpdateNotSupported(t *testing.T) {
 			if resp.Type != JSApiStreamUpdateResponseType {
 				t.Fatalf("Invalid response type %s expected %s", resp.Type, JSApiStreamUpdateResponseType)
 			}
-			if !IsNatsErr(resp.Error, JSStreamInvalidConfigF) {
-				t.Fatalf("Expected error regarding config error, got %+v", resp.Error)
+			if IsNatsErr(resp.Error, JSStreamInvalidConfigF) {
+				t.Fatalf("Expected no error regarding config error, got %+v", resp.Error)
 			}
 		}
-		expectFailUpdate()
 
-		// Now try with a new stream with RePublish present and then try to change config
-		cfg = &StreamConfig{
-			Name:     stream + "_2",
-			Storage:  MemoryStorage,
-			Replicas: replicas,
-			RePublish: &RePublish{
-				Source:      ">",
-				Destination: "bar.>",
-			},
+		expectRepublished := func(expectedRepub bool) {
+			t.Helper()
+
+			nc, js := jsClientConnect(t, s)
+			defer nc.Close()
+
+			// Accessed atomically to prevent races between the two
+			// subscribers in their own goroutines.
+			var foos, bars int32
+
+			// Create a subscriber for foo.> so that we can see
+			// our published message being echoed back to us.
+			_, err := nc.Subscribe(
+				"foo.>",
+				func(msg *nats.Msg) { atomic.AddInt32(&foos, 1) },
+			)
+			require_NoError(t, err)
+
+			// Create a subscriber for bar.> so that we can see
+			// any potentially republished messages.
+			_, err = nc.Subscribe(
+				"bar.>",
+				func(msg *nats.Msg) { atomic.AddInt32(&bars, 1) },
+			)
+			require_NoError(t, err)
+
+			// Publish a message, it will hit the foo.> stream and
+			// may potentially be republished to the bar.> stream.
+			_, err = js.Publish("foo."+stream, []byte("HELLO!"))
+			require_NoError(t, err)
+
+			// Wait for a little while so that we have enough time
```

Comment:
This works. One trick we use elsewhere is to make these subs sync, and then use test helper checkSubPending() which will check for a while that the number of queued messages is what we expect. Might want to consider that pattern here as well, but again this works too, but has a non-deterministic time.Sleep().

Reply:
Also then don't need atomics, callbacks with inc etc.

Reply:
That's certainly a nice helper, have refactored to use it.